### PR TITLE
Deduplicate policies prior to generating ACL on request

### PR DESCRIPTION
**Pull request by davidadeleon**
This PR solves an issue where a token can have duplicate policies as part of its policy set when using external groups. This scenario is encountered when an external group policy mapping is created in Vault containing duplicate policies to those defined by the authentication method.

### CODE REVIEW

**Review by mpalmi — COMMENTED** [review]
Would it be a good idea to add a go test for this to ensure the behavior doesn't regress?

**Review by jasonodonnell — APPROVED** [review]
LGTM, but we should try to figure out a way to add a test for this. We can sync on this.

**Review by hghaf099 — APPROVED** [review]

**Comment by hghaf099** [comment]
This PR probably needs to get backported to previous versions.

**Comment by hghaf099** [comment]
Also, I am wondering if we could go back to 1.10, and before the fix for Paxos. Apply this change, and see if we could reproduce Paxos issue. It would be a good test IMHO.

File: vault/request_handling.go [hunk 1]
```diff
@@ -204,7 +204,7 @@ func (c *Core) fetchACLTokenEntryAndEntity(ctx context.Context, req *logical.Req
 		return nil, nil, nil, nil, ErrInternalError
 	}
 	for nsID, nsPolicies := range identityPolicies {
-		policyNames[nsID] = append(policyNames[nsID], nsPolicies...)
+		policyNames[nsID] = policyutil.SanitizePolicies(append(policyNames[nsID], nsPolicies...), false)
```
[the remark this answers is not in the document]
Comment (davidadeleon) [reply vault/request_handling.go:207]: It looks like this isn't just isolated to group policy mapping, but also entity policy mapping. Without this in place checking `policyNames` before generating the ACL shows duplicated policies in a namespace ID. Ex: `map[root:[default secret secret]]`.

[the remark this answers is not in the document]
Comment (ncabatoff) [reply vault/request_handling.go:207]: Ah, I missed that we assigned to the map twice in that func, ty.

**Review by ncabatoff — APPROVED** [review]

**Pull request merged**